modules/percolator/src/main/java/org/elasticsearch/percolator/ExtractQueryTermsService.java

```diff
@@ -281,10 +305,31 @@ public static Query createQueryTermsQuery(IndexReader indexReader, String queryM
         return new TermsQuery(extractedTerms);
     }
```

Comment:
do we want to add a case for MatchAllDocsQuery now that we are adding this ability to skip the MemoryIndex?